### PR TITLE
update camera module api

### PR DESCRIPTION
**Pull request by sunqirui1987**

### CODE REVIEW

File: camera.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package spx
+
+import (
+	"github.com/goplus/spx/internal/camera"
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+type Camera struct {
+	camera.FreeCamera
```
Comment (xushiwei) [inline camera.go:9]: FreeCamera may have public method. So don't embed FreeCamera, use a member name.

File: camera.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	g  *Game
+	on interface{}
```
Comment (xushiwei) [inline camera.go:11]: use on_ because in Go+ on is public with On

File: camera.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func NewCamera(g *Game, winW, winH float64, worldW, worldH float64) *Camera {
+	c := &Camera{}
+	c.g = g
+	c.FreeCamera = *camera.NewFreeCamera(winW, winH, worldW, worldH)
+	return c
+}
+func (c *Camera) GetXY() (float64, float64) {
```
Comment (xushiwei) [inline camera.go:20]: don't provide GetXY now.

File: camera.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	cx := c.GetPos().X
+	cy := c.GetPos().Y
+	return cx, cy
+}
+func (c *Camera) SetXY(x float64, y float64) {
```
Comment (xushiwei) [inline camera.go:25]: MoveTo is better. or use SetXYpos (same as Sprite)

File: camera.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+type Camera struct {
+	camera.FreeCamera
+	g  *Game
+	on interface{}
+}
+
+func NewCamera(g *Game, winW, winH float64, worldW, worldH float64) *Camera {
```
Comment (xushiwei) [inline camera.go:14]: this should be a private func.

File: game.go [hunk 6]
```diff
@@ -66,13 +66,21 @@ type Game struct {
...
+	//world
+	worldWidth_  int
+	worldHeight_ int
+	stepUnit     float64 //global step unit in game
 
 	gMouseX, gMouseY int64
 
-	sinkMgr eventSinkMgr
+	sinkMgr   eventSinkMgr
+	isStopped bool
```
Comment (xushiwei) [inline game.go:80]: this member is removed.

File: camera.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+func (c *Camera) GetXY() (float64, float64) {
+	cx := c.GetPos().X
+	cy := c.GetPos().Y
+	return cx, cy
+}
+func (c *Camera) SetXY(x float64, y float64) {
+	c.on = nil
+	c.MoveTo(x, y)
+}
+func (c *Camera) Move(x float64, y float64) {
```
Comment (xushiwei) [inline camera.go:29]: use ChangeXYpos (same as Sprite)?

File: test/Camera/Monkey.spx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+onKey KeySpace, => {
+	Game.Camera.On("Monkey")
+}
+onKey KeyB, => {
+	step 10
+}
+onKey KeyUp, => {
+	x,y := Game.Camera.GetXY()
+	y = y + 5.0
+	Game.Camera.SetXY(x,y)
```
Comment (xushiwei) [inline test/Camera/Monkey.spx:17]: why not `Camera.changeXYpos 0, 0.5`?

File: test/MiniMapCamera/Monkey.spx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	animate "think"
+}
+
+onKey KeyC, => {
+	turn 45
+}
+
+onKey KeyE, => {
+	step 5
+	println distanceStepTo("Crocodile")
```
Comment (xushiwei) [inline test/MiniMapCamera/Monkey.spx:19]: what is `distanceStepTo`?

File: test/MiniMapCamera/res/sounds/clap/index.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "path": "monkey_clap.mp3"
```
Comment (xushiwei) [inline test/MiniMapCamera/res/sounds/clap/index.json:2]: may share mp3 in `https://github.com/goplus/spx/tree/main/test/res`?

File: game.go (hunk 6 above)
Comment (sunqirui1987) [reply game.go:80]: this member  exist

**Pull request merged**